### PR TITLE
fix (dashboard): improve project health error handling, add poll interval

### LINKED ISSUE #2776: dashboard: overview project health modal doesn't show error info
Service state modal has a transient state where it might show the error `undefined is not an object (evaluating r.length)` instead of the expected accordion for each service

### PR DESCRIPTION
**Pull request by dbm03**
Fixes #2776

### CODE REVIEW

File: dashboard/src/features/projects/common/hooks/useServiceStatus/useServiceStatus.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+      skip: !isPlatform || !currentProject,
+      pollInterval: 0,
+      skipPollAttempt: () => !isVisible,
+    });
+
+  // Fetch when mounted
+  useEffect(() => {
+    refetch();
+    return () => {
+      stopPolling();
```
Comment (onehassan) [inline dashboard/src/features/projects/common/hooks/useServiceStatus/useServiceStatus.ts:43]: I think this is redundant here, see Line 77

Comment (dbm03) [reply dashboard/src/features/projects/common/hooks/useServiceStatus/useServiceStatus.ts:43]: Not sure, if there's no Line 77, then useEffect wouldn't stop polling after cleanup (when `options.shouldPoll=true`), right?

Maybe it's clearer to rewrite as:
```
useEffect(() => {
    if (options.shouldPoll) {
      if (
        [auth, storage, postgres, hasura, ai, ...Object.values(run)].some(
          (service) => service?.state !== ServiceState.Running,
        )
      ) {
        // Poll more frequently if a service is not running
        startPolling(options.pollInterval || 10000);
      } else {
        // Poll less frequently when all services are running
        startPolling(60000);
      }
    }
    

    return () => stopPolling();
  }
  ```

Comment (onehassan) [reply dashboard/src/features/projects/common/hooks/useServiceStatus/useServiceStatus.ts:43]: in that case the cleanup will be done on L43 right — That will be executed when the component unmounts anyway

Comment (dbm03) [reply dashboard/src/features/projects/common/hooks/useServiceStatus/useServiceStatus.ts:43]: Oh I see your point now, I'll update it :+1:

**Review by onehassan — APPROVED** [review]
LGTM

**Pull request merged**